chrome/content/zotero/xpcom/data/itemFields.js

```diff
@@ -172,7 +173,7 @@ Zotero.ItemFields = new function() {
 		}
 		
 		if (_itemTypeFields[itemTypeID]) {
-			return _itemTypeFields[itemTypeID];
+			return _itemTypeFields[itemTypeID].slice();
```

Comment:
Honestly, I don't exactly recall, but that sounds like the concern I would have had. Looking through the code, I don't see anything that attempts to modify the returned value, so freezing it should be OK. (I've actually never come across `Object.freeze`, which is rather neat)